### PR TITLE
docs: update documentation to account for parquet source

### PR DESCRIPTION
**Pull request by Meghajit**
Closes https://github.com/odpf/dagger/issues/189

### CODE REVIEW

File: docs/docs/concepts/architecture.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ Dagger or Data Aggregator is a cloud native framework for processing real-time s
```
Comment (prakharmathur82) [inline docs/docs/concepts/architecture.md:6]: Few general feedbacks:
* you will need to update readme as
* in the readme, and actually everywhere else, we are saying dagger is a real-time stream processing tool, we need to change that, maybe real-time and historical data processing?
* let's add a separate sub section under architecture section called Source, we can add details for Kafka and Parqeut and later BQ source can also come, as of now there is no starting point of our documentation for parquet source?

**Review by prakharmathur82 — CHANGES_REQUESTED** [review]

File: docs/docs/concepts/architecture.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ Dagger or Data Aggregator is a cloud native framework for processing real-time s
 
 ### Components
 
-![Dagger Architecture](/img/architecture.png)
+![Dagger Architecture](/img/system_design/dagger_system_design.png)
```
[the remark this answers is not in the document]
Comment (Meghajit) [reply docs/docs/concepts/architecture.md:9]: But the image is getting rendered in the page. Do you mean you are not able to open the image via the link ?

Screenshot 2022-08-18 at 11 58 47 AM

File: docs/static/img/system_design/dagger_system_design.drawio [hunk 3]
```diff
@@ -0,0 +1 @@
...4tEKb22AJzGPCcpfbNZqslJPB6Rwu85dphvLcYlnp9rmpEgYai7e8+GAw6ilCnBZrjZTFnL4qxGkxdXz6KqSuN2iuT0CR7w23QHGsngsUp4eiVC3CertVWC6j6CS8alp7bInChJgEHsodaRGzdu7Iuv7DZs4TGKZEojdXfNoizorJs25rT0cWIxhe8jj75vFyFPNmHeMlawW363yVP8zb3sAMbnQhnKDhrQz1xQUMgezNjmVsv0Xxq3tdx4UpLJk1TUVGqNKlLW08A21WdhzvmF2gyP564fPgqndU+waBrcs7VovJ654Wb4AW6kPH3jLlPnYDZ5ttIMZ9kXCd1Hl148FRu1v3czYbPgQFRbHoQtWrHU1AKObsSiBuUEqqZDJM/zr19pzBwqHagazEw7vUhYjpKiqrLjWuyyFt2Nk9EjXvMKYoNW/163URArvNSi/8M16u8iLufjJHC/hWup/RrvvpG8UVEuTD59t3H8+oVtWHllD5AEVOqkoj9FU+vixSlvtKzwUuHaXqYKQEH9QD9Ja1hKFoB8DoZSPkoasI2mvGw3vqgN02FQG2fWQuUrfIyeHCt9fyYRaNF3I9KRvWvra2153wh60KPzXVa9ZadPPkHqstmY+eUq7hIE9Bq+M5DVIonjsBz+RJDCzHHArTpCl33UxT7BCTqNe3ii/J1Y5qAUO3EmOZNA0jtTmT7tZ+TRvpCUf2t+PgVWzxOz8Jualb2XzwcRSzY53KxeUm7FegF+/pe8/qXRomF/8d0g4IaXbrkAb6DWnyIyHOTgbes1e5ikShU8UXAFl7LyXfc5GoToKMoO56iqkwti2mG21KsTNznGASnYDxHsbEV+c/ygjrfkHbJ0bpG2HkCZBX7nPX7T2nCiSfnrUGWyNMHSJedJQggNYJgp6NrOJzhj3x8ebtpVn+OuJGH91ihf3KphNqqHxK8tltydfz7pzyRpUF9+sp/DxB8HMUawu7rvvsfqY7NG5762/qgk6vtZqI67cctOVy8LXlYo6oG/vmJeg7H7NfwWqs/dzutnxgm74q3Jd9LnmZbemcwgym+XaGZl8f9VM5JRJ8XJqiJ7ja7EPt9NdyxIcCqB74o9ooIT726oSY7fe8Nup0LftD248PzVH/917QJtxVD6v53Vd39FXHkMe0E/VVcrj9gbMC4u3PxFm3/wM=</diagram></mxfile>
```
[the remark this answers is not in the document]
Comment (Meghajit) [reply docs/static/img/system_design/dagger_system_design.drawio:1]: > why is source connected to proto handler?

Hmm. I guess the source should only be connected to the deserializer. The deserializer and proto handler work together to parse raw data ( parquet records, kafka records) into Row. Will fix this

> why call it proto handler?

Yes, I should have called it `Type Handler`. It was a mistake during copy paste from the earlier diagram. I will edit the name.

> again check for pre processor workflow

Will do this.

> same for name of proto handler towards sink

Will change it to `Type Handler`

File: docs/docs/concepts/architecture.md (hunk 1 above)
Comment (Meghajit) [reply docs/docs/concepts/architecture.md:6]: > in the readme, and actually everywhere else, we are saying dagger is a real-time stream processing tool, we need to change that, maybe real-time and historical data processing?

I suppose this was done for the other sections in the earlier PR (https://github.com/odpf/dagger/pull/153). I will update the README and other places where not done.

> let's add a separate sub section under architecture section called Source, we can add details for Kafka and Parqeut and later BQ source can also come, as of now there is no starting point of our documentation for parquet source?

Makes sense. Let's add a page called `Configure a Source` where we can mention all the details about the sources. Then we can link this page in the `Create a Job` page under [this section](https://odpf.github.io/dagger/docs/guides/create_dagger#a-source)

File: docs/docs/concepts/lifecycle.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 Architecturally after the creation of Dagger, it goes through several stages before materializing the results to an output stream.
 
-![](/img/dagger-lifecycle.png)
+![](/img/lifecycle/dagger_lifecycle.png)
```
[the remark this answers is not in the document]
Comment (Meghajit) [reply docs/docs/concepts/lifecycle.md:5]: Page is getting rendered though

File: docs/docs/intro.md [hunk 5]
```diff
@@ -9,7 +9,7 @@ for stateful processing of both real time and historical streaming data. With Da
 applications or complicated code to process data as a stream. Instead, you can write SQL queries and UDFs to do the 
 processing and analysis on streaming data.
 
-![](/img/overview.svg)
+![](/img/overview/dagger_overview.png)
```
[the remark this answers is not in the document]
Comment (Meghajit) [reply docs/docs/intro.md:12]: Image is getting rendered though

File: docs/static/img/lifecycle/dagger_lifecycle.drawio [hunk 6]
```diff
@@ -0,0 +1 @@
...>7Vldd5owGP41XrYHiPhx2YndutOdtceuay8ziJA2Ek8SK+7XL0gQMNG5HTXsrDet75MAyfO8X4EOGM2yjwzOky80QqTjOVHWAUHH8zzHBfJfjqwKpDfwCiBmOCogtwIm+CdSoKPQBY4Qb0wUlBKB500wpGmKQtHAIGN02Zw2paT51DmMkQZMQkh09DuORFKgA69f4Z8QjpPyyW5vWIzMYDlZ7YQnMKLLGgTGHTBilIri1ywbIZKTV/JSXHe9Y3SzMIZSccgFr8uHIBPPF59vHkL42Hv5+iN7vPCUGlysyh2jSBKgTMpEQmOaQjKu0A+MLtII5bd1pFXNuaV0LkFXgi9IiJVSEy4ElVAiZkSNyhWz1ZO6fm0858alX5pBVh8MVsrigtFXNKKEsvVaAQDD4bXc8IdiF/nSd7KjIE4XLER7KCm9DLIYiT3z+hsNpfMjOkNypfI6hggU+K25Dqi8MN7Mq4SSP5RWf6Cbuu8bJAv1JE3HpkrLBAs0mcP11pcyVpuKqNshJlC2n0F9x+UFZciWka7MZRU2rq+wpBYyoJx4dJI2kr8794aSA517h9TncW6va1e3SqrnhlJm3VAaXeWFRpohgZzjcC0mZMIEt0hjYFVjLYEFMI4Rk9ik2KDtfHbhOs2EtulJ6hnNM2S0kyU0sJu0AHHEMCTSu1n7qANd29T5GnUPDKZ8StkMMW6dMdfrX/pNynwDZa6vU9Y9WfkE7+XT1PMdkFoHNlNrf09qvb+Vf+8XSD3GpssD0Lb06vbfHd7kyIf0i1abiYHm8da92/fadhpye1a8O8Piqfa75tvSqlw7N1Z1P/+LiGhdJz48NHp8m9Ez1KJnnAnEpMr2owhst0WeqZM8a1tUlqkaXzdpa/kCjnW+9KNeq1pvA2emZH1ezvac9CY4fdUzOSF4ztHvCYN8Xrwmn+IsJ/kYDPZ7W52cwecGhlo3OBl9+mmvSmlOAAVsz2uGfnfrqGw69523DdbbqTsmiXLuGA0R59QQtHL/oklNs8CmNM2dc4oJ2YIgwXGal2aUJ1EJ5GziEJIrNTDDUUR2eXZTrSPI4Xa97Yrj+bo3G5IBOJkeeoG+o1z8J4L43e307BpagOE5BfH144XF99ObMZvnRP+f+GpWrtJUVr8F1/Z7ke13IsZO5EjFQJrVx+f1WO0TPhj/Ag==</diagram></mxfile>
```
[the remark this answers is not in the document]
Comment (Meghajit) [reply docs/static/img/lifecycle/dagger_lifecycle.drawio:1]: I will  check this

File: docs/docs/concepts/architecture.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (prakharmathur82) [reply docs/docs/concepts/architecture.md:9]: I meant I can't see here : https://github.com/Meghajit/dagger/blob/feat/dstrm%231009-update-dagger-documentation/docs/docs/concepts/architecture.md

[the remark this answers is not in the document]
Comment (Meghajit) [reply docs/docs/concepts/architecture.md:9]: Yeah, images don't get rendered on the markdown page for some reason. It was happening earlier also. You can check the master branch: https://github.com/odpf/dagger/blob/main/docs/docs/concepts/architecture.md

[the remark this answers is not in the document]
Comment (Meghajit) [reply docs/docs/concepts/architecture.md]: Updated the relative paths
Fixed via commit https://github.com/odpf/dagger/pull/190/commits/2de83600e20c2a53667a2acb529c23fae09dbb3b

File: docs/docs/concepts/lifecycle.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Meghajit) [reply docs/docs/concepts/lifecycle.md]: Updated the relative paths
Fixed via commit https://github.com/odpf/dagger/pull/190/commits/2de83600e20c2a53667a2acb529c23fae09dbb3b

File: docs/docs/intro.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Meghajit) [reply docs/docs/intro.md]: Updated the relative paths
Fixed via commit https://github.com/odpf/dagger/pull/190/commits/2de83600e20c2a53667a2acb529c23fae09dbb3b

File: docs/static/img/lifecycle/dagger_lifecycle.drawio (hunk 6 above)
[the remark this answers is not in the document]
Comment (Meghajit) [reply docs/static/img/lifecycle/dagger_lifecycle.drawio:1]: Verified, preprocessors operate on Rows, and hence operate after data is deserialized

File: docs/static/img/system_design/dagger_system_design.drawio (hunk 3 above)
[the remark this answers is not in the document]
Comment (Meghajit) [reply docs/static/img/system_design/dagger_system_design.drawio]: Updated the diagrams
Fixed via commit https://github.com/odpf/dagger/pull/190/commits/7242b10a0b4957cd33f67cfab9488355e1fa9c4d

File: docs/docs/concepts/architecture.md (hunk 1 above)
Comment (Meghajit) [reply docs/docs/concepts/architecture.md:7]: Fixed via commit https://github.com/odpf/dagger/pull/190/commits/eb9cc0bd3f455e661ab4687ad255ebbeacaff6ba and https://github.com/odpf/dagger/pull/190/commits/4fe1e05f3d368377b3f90d9d27efc1e1af69224e

File: docs/docs/guides/create_dagger.md [hunk 7]
```diff
@@ -12,73 +12,8 @@ Dagger is a stream processing framework built with Apache Flink to process/aggre
...
-        <name>fs.gs.requester.pays.project.id</name>
-        <value>my_billing_project_id</value>
-        <final>true</final>
-    </property>
-</configuration>
-```
-You can look into the official [GCS Hadoop Connectors](https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/master/gcs/CONFIGURATION.md) 
-documentation to know more on how to edit this xml as per your needs.
+Dagger requires configuring a source from where data will be streamed for processing. Please check 
+[here](./guides/choose_source.md) for the different available data sources.
```
Comment (prakharmathur82) [inline docs/docs/guides/create_dagger.md:16]: link is broken

File: docs/docs/guides/choose_source.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+        <final>true</final>
+    </property>
+    <property>
+        <name>fs.gs.requester.pays.project.id</name>
+        <value>my_billing_project_id</value>
+        <final>true</final>
+    </property>
+</configuration>
+```
+
```
Comment (prakharmathur82) [inline docs/docs/guides/choose_source.md:78]: let's add link to source config reference?

File: docs/docs/guides/overview.md [hunk 9]
```diff
@@ -2,6 +2,11 @@
 
 The following section describes how to manage Dagger throughout its lifecycle.
 
+### [Choosing a Dagger Source](./guides/choose_source.md)
```
Comment (prakharmathur82) [inline docs/docs/guides/overview.md:5]: link is broken

**Review by prakharmathur82 — CHANGES_REQUESTED** [review]

File: README.md [hunk 10]
```diff
@@ -4,19 +4,21 @@
 [![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg?logo=apache)](LICENSE)
 [![Version](https://img.shields.io/github/v/release/odpf/dagger?logo=semantic-release)](https://github.com/odpf/dagger/releases/latest)
 
-Dagger or Data Aggregator is an easy-to-use, configuration over code, cloud-native framework built on top of Apache Flink for stateful processing of real-time streaming data. With Dagger, you don't need to write custom applications or manage resources to process data in real-time.
-Instead, you can write SQLs to do the processing and analysis on streaming data.
+Dagger or Data Aggregator is an easy-to-use, configuration over code, cloud-native framework built on top of Apache Flink
+for stateful processing of both real time and historical streaming data. With Dagger, you don't need to write custom
```
[the remark this answers is not in the document]
Comment (Meghajit) [reply README.md:8]: @prakharmathur82 We are able to process both real time + historical batched data ( parquet files) as a stream. Hence, put the streaming keyword after. Does the below look ok ?

`Dagger or Data Aggregator is an easy-to-use, configuration over code, cloud-native framework built on top of Apache Flink for stateful processing of streaming data, both real time and historical.`

File: README.md [hunk 11]
```diff
@@ -4,19 +4,21 @@
...
 Discover why to use Dagger
 
-* **Processing:** Dagger can transform, aggregate, join and enrich Protobuf data in real-time.
-* **Scale:** Dagger scales in an instant, both vertically and horizontally for high performance streaming sink and zero data drops.
-* **Extensibility:** Add your own sink to dagger with a clearly defined interface or choose from already provided ones.
-* **Pluggability:** Add custom business logic in form of plugins \(UDFs, Transformers, Preprocessors and Post Processors\) independent of the core logic. 
-* **Metrics:** Always know what’s going on with your deployment with built-in [monitoring](https://odpf.github.io/dagger/docs/reference/metrics) of throughput, response times, errors and more.
+- **Processing:** Dagger can transform, aggregate, join and enrich streaming data, both real-time and historical.
+- **Scale:** Dagger scales in an instant, both vertically and horizontally for high performance streaming sink and zero data drops.
+- **Extensibility:** Add your own sink to dagger with a clearly defined interface or choose from already provided ones. Use Kafka and/or Parquet Files as stream sources.
```
[the remark this answers is not in the document]
Comment (Meghajit) [reply README.md:19]: @prakharmathur82 You can create a Dagger which consumes from Parquet in one stream and Kafka in another. It's just a source :-)

File: docs/docs/concepts/architecture.md [hunk 12]
```diff
@@ -1,12 +1,13 @@
 # Architecture
 
-Dagger or Data Aggregator is a cloud native framework for processing real-time streaming data built on top of Apache Flink.
+Dagger or Data Aggregator is a cloud native framework built on top of Apache Flink for processing both real-time and 
+historical streaming data.
```
[the remark this answers is not in the document]
Comment (Meghajit) [reply docs/docs/concepts/architecture.md:4]: Commented above for the same

File: README.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (ravisuhag) [reply README.md:8]: We can just say "stateful processing of data"

File: docs/docs/guides/overview.md (hunk 9 above)
Comment (Meghajit) [reply docs/docs/guides/overview.md]: Seems like these links are broken in the .md pages even in [master](https://github.com/odpf/dagger/blob/main/docs/docs/guides/overview.md) branch.

Fixed via commit https://github.com/odpf/dagger/pull/190/commits/49eca4140cc54d6968fec1f6980ea702d78760fd

File: docs/docs/guides/create_dagger.md (hunk 7 above)
Comment (Meghajit) [reply docs/docs/guides/create_dagger.md]: Fixed via commit https://github.com/odpf/dagger/pull/190/commits/5accebb52f3d4f92d9517ba804636ee2532facc6

File: README.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (prakharmathur82) [reply README.md:8]: we can say stateful processing of data

**Comment by ravisuhag** [comment]
@Meghajit can you cross check if docs build is passing by running  `yarn build` in `/docs` folder.

File: docs/docs/guides/choose_source.md (hunk 8 above)
Comment (Meghajit) [reply docs/docs/guides/choose_source.md:78]: Fixed via commit 
https://github.com/odpf/dagger/pull/190/commits/1b40318b4683998e3425d6842f26165c5ce4c4df

File: README.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (Meghajit) [reply README.md:8]: Cool 👍

[the remark this answers is not in the document]
Comment (Meghajit) [reply README.md]: Updated via commit https://github.com/odpf/dagger/pull/190/commits/ab4a27218b66ec057735e1ed86b655637c6c7547

File: docs/docs/concepts/architecture.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (Meghajit) [reply docs/docs/concepts/architecture.md]: Fixed via commit https://github.com/odpf/dagger/pull/190/commits/f7bb8b08096be16a9cb800306088e075ff58aa0f

**Comment by Meghajit** [comment]
> @Meghajit can you cross check if docs build is passing by running `yarn build` in `/docs` folder.
Screenshot 2022-08-23 at 11 32 08 AM

@ravisuhag Yes, it passed

**Review by prakharmathur82 — APPROVED** [review]

**Pull request merged**